doc/meetings/2022-01-11.md

```diff
@@ -0,0 +1,37 @@
+# Node.js  Loaders Team Meeting 2022-01-11
+
+## Links
+
+* **Recording**: https://www.youtube.com/watch?v=js_NQp-bomM
+* **GitHub Issue**: https://github.com/nodejs/loaders/issues/58
+
+## Present
+
+* Geoffrey Booth: @GeoffreyBooth
+* Bradley Farias: @bmeck
+* Jacob Smith: @jakobjingleheimer
+* Maël Nison: @arcanis
+
+## Agenda
+
+## Announcements
+
+*Extracted from **loaders-agenda** labeled issues and pull requests from the **nodejs org** prior to the meeting.
+
+### nodejs/node
+
+* esm: implement the getFileSystem hook [#41076](https://github.com/nodejs/node/pull/41076)
+
+### nodejs/loaders
+
+* Loader chaining *while loading loaders* [#56](https://github.com/nodejs/loaders/issues/56)
+
+
+## Notes
+
+Next steps:
+* Maël will write a design doc for filesystem hooks, submit as a PR in the loaders repo.
+* Before filesystem looks land, two other PRs need to precede it:
+  * Chaining loaders
+  * Moving loaders off-thread (or could possibly go after filesystem hooks)
```

Comment:
```suggestion
  * Moving loaders off-thread (or could possibly go after filesystem hooks)
* Maël mentioned Yarn PnP will need the ability to implement other loaders (where yarn will act as a loader of loaders). After discussing concerns raised by Bradley and Jacob around the significant logical complexity implications as well as the worker context issues (not all workers will have all the same loaders), we determined the use-case to likely to need merely a higher-tier of loaders rather than special loaders within the same tier; we've dubbed this higher-tier "ambient loaders" (which would have a different CLI flag, ex `--ambient-loader=./godly-loader.mjs --loader=./ts-loader.mjs`). Leveraging this will however still mean the system is non-deterministic, and thus allowing for high probability of foot-gunning.
  * Maël will write up a proposal doc, including a thorough explanation of the use-case.
```